src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -327,12 +327,7 @@ _test_session_pool_reap (bool pooled)
     */
    b->server_session->last_used_usec = bson_get_monotonic_time ();
    mongoc_client_session_destroy (b);
-   BSON_ASSERT (client->topology->session_pool);
-   ASSERT_SESSIONS_MATCH (&lsid_b, &client->topology->session_pool->lsid);
```

Comment:
Instead of removing the assertion of lsid, can this use `mongoc_ts_pool_try_pop` to get session B, check the resulting `lsid`, then assert that the session pool is empty?